tools/code_format/check_format.py

```diff
@@ -361,8 +361,8 @@ def allowlistedForUnpackTo(file_path):
 
 
 def denylistedForExceptions(file_path):
-  return (file_path in EXCEPTION_DENYLIST or isInSubdir(file_path, 'tools/testdata')) and \
-      not file_path.endswith(DOCS_SUFFIX)
+  return (file_path.endswith(".h") and not file_path.startswith("./test/")) or (file_path in EXCEPTION_DENYLIST \
+      or isInSubdir(file_path, 'tools/testdata')) and not file_path.endswith(DOCS_SUFFIX)
```

Comment:
not strictly necesary, but can we re-order/format this or comment to make it a little more readable? 
`# Include source headers, EXCEPTION_DENYLIST, and test data`. 

The `DOCS_SUFFIX` applies because `isInSubdir` will also include all root files like docs suffices. So maybe these are a bundled group too.

Reply:
I agree the logic is a little convoluted. It is essentially (is header && not in test/) || ((is in exception denyList or in tool/testdata) && not a doc). I don't necessarily see a clear intersection here, but I can certainly try to improve the readability by commenting and may be assigning them different variable names.

Reply:
What do you think about a comment and then a small restructure like this:

if doc: 
   return False
if (is header && not in test) or (is in exceptionDenyList) or (is in tools/testdata): 
   return True

Reply:
yes. That looks pretty nice!